packages/capacitor-plugin/src/definitions.ts

```diff
@@ -0,0 +1,565 @@
+   *
+   * @default false
+   * @since 1.0.0
+   */
+  recursive?: boolean;
+}
+
+export interface RmdirOptions {
+  /**
+   * The path of the directory to remove
+   *
+   * @since 1.0.0
+   */
+  path: string;
+
+  /**
+   * The `Directory` to remove the directory from
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+
+  /**
+   * Whether to recursively remove the contents of the directory
+   *
+   * @default false
+   * @since 1.0.0
+   */
+  recursive?: boolean;
+}
+
+export interface ReaddirOptions {
+  /**
+   * The path of the directory to read
+   *
+   * @since 1.0.0
+   */
+  path: string;
+
+  /**
+   * The `Directory` to list files from
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+}
+
+export interface GetUriOptions {
+  /**
+   * The path of the file to get the URI for
+   *
+   * @since 1.0.0
+   */
+  path: string;
+
+  /**
+   * The `Directory` to get the file under
+   *
+   * @since 1.0.0
+   */
+  directory: Directory;
+}
+
+export interface StatOptions {
+  /**
+   * The path of the file to get data about
+   *
+   * @since 1.0.0
+   */
+  path: string;
+
+  /**
+   * The `Directory` to get the file under
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+}
+
+export interface CopyOptions {
+  /**
+   * The existing file or directory
+   *
+   * @since 1.0.0
+   */
+  from: string;
+
+  /**
+   * The destination file or directory
+   *
+   * @since 1.0.0
+   */
+  to: string;
+
+  /**
+   * The `Directory` containing the existing file or directory
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+
+  /**
+   * The `Directory` containing the destination file or directory. If not supplied will use the 'directory'
+   * parameter as the destination
+   *
+   * @since 1.0.0
+   */
+  toDirectory?: Directory;
+}
+
+export type RenameOptions = CopyOptions;
+
+export interface ReadFileResult {
+  /**
+   * The representation of the data contained in the file
+   *
+   * Note: Blob is only available on Web. On native, the data is returned as a string.
+   *
+   * @since 1.0.0
+   */
+  data: string | Blob;
+}
+
+export interface WriteFileResult {
+  /**
+   * The uri where the file was written into
+   *
+   * @since 1.0.0
+   */
+  uri: string;
+}
+
+export interface ReaddirResult {
+  /**
+   * List of files and directories inside the directory
+   *
+   * @since 1.0.0
+   */
+  files: FileInfo[];
+}
+
+export interface FileInfo {
+  /**
+   * Name of the file or directory.
+   */
+  name: string;
+  /**
+   * Type of the file.
+   *
+   * @since 1.0.0
+   */
+  type: 'directory' | 'file';
+
+  /**
+   * Size of the file in bytes.
+   *
+   * @since 1.0.0
+   */
+  size: number;
+
+  /**
+   * Time of creation in milliseconds.
+   *
+   * It's not available on Android 7 and older devices.
+   *
+   * @since 1.0.0
+   */
+  creationTime?: number;
+
+  /**
+   * Time of last modification in milliseconds.
+   *
+   * @since 1.0.0
+   */
+  modificationTime: number;
```

Comment:
This is a breaking change. I understand that these parameter names are more readable, but I'm not sure if it's worth causing a breaking change over just this.

```suggestion
  /**
   * Time of creation in milliseconds.
   *
   * It's not available on Android 7 and older devices.
   *
   * @since 4.0.0
   */
  ctime?: number;

  /**
   * Time of last modification in milliseconds.
   *
   * @since 4.0.0
   */
  mtime: number;
```

We could instead deprecate the `ctime` and `mtime` properties and have their values be mapped from `creationTime` and `modificationTime`, or something along those lines?

Reply:
what happens when the OS asset moves to use the capacitor bridge?

Reply:
What you mean is that `ctime` and `mtime` will not match with the current cordova plugin, causing Synapse to fail.

There's 3 options that I see:

1. We would change the definition on cordova plugin / usage on wrapper to match the current capacitor definition
2. We cause a breaking change in the capacitor plugin by replacing the names with `creationTime` `modificationTime`
3. We keep the current names, deprecate them, and add the new ones as well

Unsure if option 1 is possible, but I think it is. Option 2 might cause "friction". Option 3 only requires changes in the capacitor plugin.

For Option 3, one could eventually cause the breaking change when Cap 8 comes around (assuming that convention keeps being followed). Although, I think the `ctime` and `mtime` names are meant to resemble that of the web API - as the current plugin description suggests: "The Filesystem API provides a NodeJS-like API for working with files on the device."

Your thoughts?

Reply:
Option 1 should be possible, but we need to be careful in mapping the json to the OS structure: on the OS side, we're better of leaving the `creationTime` and `modificationTime`

Reply:
changed it back to `mtime` and `ctime`

Reply:
Agreed, thank you! 🙇